### PR TITLE
Integrate feature branch for v2 De-Risking Guide

### PR DESCRIPTION
**Pull request by nateborr**
## Changes proposed in this pull request:

Publish the De-risking Guide 2.0 and update the links to the de-risking guide within other guide pages.

_The major changes in this PR have been incrementally reviewed and merged into the `derisking-government-tech` feature branch._ See the list of PRs below. For some small content issues that shook out of our final review, I've made fixes directly on the feature branch.

@edwintorres @lalitha-jonnalagadda @quarterback For content review of the lightweight changes in other guides to link to the 2.0 de-risking guide, see "Content changes in other guides" below.

[PR build](https://federalist-8bc0b42c-5fd0-4ae8-9366-cd9c265a4446.sites.pages.cloud.gov/preview/18f/guides/derisking-government-tech/derisking-government-tech/)

### PRs merged into the feature branch

Note that I've squashed all commits and rebased onto the main branch.

- https://github.com/18F/guides/pull/665
- https://github.com/18F/guides/pull/672
- https://github.com/18F/guides/pull/675
- https://github.com/18F/guides/pull/677
- https://github.com/18F/guides/pull/679
- https://github.com/18F/guides/pull/681
- https://github.com/18F/guides/pull/680
- https://github.com/18F/guides/pull/684
- https://github.com/18F/guides/pull/692
- https://github.com/18F/guides/pull/693

### New guide content

The De-risking Guide 2.0 can be reached at the Guides domain's [`/derisking-government-tech/`](https://federalist-8bc0b42c-5fd0-4ae8-9366-cd9c265a4446.sites.pages.cloud.gov/preview/18f/guides/derisking-government-tech/derisking-government-tech/) route.

### v1 de-risking guide redirect banners

This change adds a banner message to the top of the original de-risking guide pages, informing readers of the 2.0 guide and providing a link from each v1 page to the corresponding v2 page.

### Content changes in other guides

The changes in other guides are lightweight updates to de-risking guide links and references, so that they point to the 2.0 guide. They were reviewed for correctness in: https://github.com/18F/guides/pull/679

#### Engineering guide

[Resources --> Related links](https://federalist-8bc0b42c-5fd0-4ae8-9366-cd9c265a4446.sites.pages.cloud.gov/preview/18f/guides/derisking-government-tech/engineering/about/resources/#related-links)

#### Methods guide

[User stories --> Additional resources](https://federalist-8bc0b42c-5fd0-4ae8-9366-cd9c265a4446.sites.pages.cloud.gov/preview/18f/guides/derisking-government-tech/methods/decide/user-stories/#add-personas)

#### Product guide

- [About this guide --> Resources --> Guides](https://federalist-8bc0b42c-5fd0-4ae8-9366-cd9c265a4446.sites.pages.cloud.gov/preview/18f/guides/derisking-government-tech/product/resources/#guides)
- [Enable partners --> Work with the vendor --> Resources](https://federalist-8bc0b42c-5fd0-4ae8-9366-cd9c265a4446.sites.pages.cloud.gov/preview/18f/guides/derisking-government-tech/product/partners/vendor/#resources) (`QASP Example [1]` link)

### Guides site updates

These updates do not change the behavior of any guides other than the de-risking guides, other than the changes to links described above. However, they do extend the site navigation configurability and make changes to shared HTML layouts. These changes have been code-reviewed in the PRs above and tested that they don't cause regressions in other guides.

- `_includes` files and layouts
  - wide layout: Add the first tag value as a class to the `main` element, as we do in the page layout, to apply custom CSS styles for a guide
  - page layout: Add an optional call-to-action above the footer for `derisking-government-tech` pages only
  - header: Add a redirect banner for `derisking` pages with a redirect URL specified
- Navigation
  - Update primary nav configuration to accept an optional `download` key, which adds the download attribute to the nav link
  - Update `secondary_navigation` configuration to accept an optional `guides` key, which takes a list of guides to render the secondary nav link for
  - Update `development.md` documentation accordingly
  - Modify the javascript that handles left-side subnav highlighting to ignore page headings without IDs (so that the new call-to-action doesn't cause errors in that functionality)

## Testing

@ameliaswong, @melchoyce, and I will jointly re-review these content changes. We've currently scheduled that review for Sept. 9.

For code review, manually smoke-test that the behavior for the new and the existing guides is in order, particularly in areas that have received code changes:

- The rendering of page headers and footers
- Primary navigation
- Secondary navigation links (i.e. "More 18F guides") in the upper right of the page header
- Shake out the redirect banners on v1 `/derisking/` pages

## security considerations

No security considerations: These updates use the existing framework and patterns of the 18F Guides site.

### CODE REVIEW

**Review by lalitha-jonnalagadda — APPROVED** [review]

**Comment by nateborr** [comment]
@ameliaswong @melchoyce I've implemented the content tune-ups we discussed in our review earlier today.

**Comment by ameliaswong** [comment]
> @ameliaswong @melchoyce I've implemented the content tune-ups we discussed in our review earlier today.

@nateborr I pinged you with a few more content tune-ups in Slack. Also completed review so that's it from me. Thank you!

**Comment by nateborr** [comment]
@ameliaswong I've completed the remaining small fixes you mentioned in Slack.

**Comment by ameliaswong** [comment]
Thank you @nateborr

**Review by quarterback — APPROVED** [review]

**Review by edwintorres — APPROVED** [review]
LGTM, I tested it locally.

There is a small console error which is not affecting the behavior or aspect of the site. It's not a blocker.

 It happen when you click a sub-menu item and then scroll the site. 

image

**Comment by nateborr** [comment]
@edwintorres I've pushed up a fix for the left-side subnav highlighting so that it no longer throws errors when you scroll the new pages that have the "need contracting help" footer on them.

File: content/derisking/qasp.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ redirect_from:
   - derisking/state-field-guide/qasp/
 layout: layouts/page
 tags: derisking
+redirect-url: /derisking-government-tech/resources/
```
Comment (adunkman) [inline content/derisking/qasp.md:8]: Pulling through one of the review items from Aug 29: 

>The redirect from the QASP page currently sends you to the Resources page, which doesn’t include the text QASP (it’s called “Quality indicators plan”). If you’re looking for the updated QASP, you may not find it. 
>
>I anticipate y’all have found “Quality indicators plan” to be more clear than QASP. We can quickly address this by adjusting the redirect to land directly on the sample. 

```suggestion
redirect-url: /derisking-government-tech/resources/quality-indicators/
```

**Review by adunkman — APPROVED** [review]
Looks good! Pulled in one suggested edit from the previous review I did, but otherwise this is good to ship tomorrow from my book.

**Pull request merged**